deploy/runtools/run_farm.py

```diff
@@ -804,3 +810,500 @@ def terminate_by_inst(self, inst: Inst) -> None:
+        with open(local_tmp_path, "w") as db_file:
+            try:
+                json.dump(host_machine_fpgas_db, db_file, indent=2)
+            except TypeError as e:
+                raise ValueError(f"Error writing JSON to {db_path}: {e}")
+        rootLogger.info(f"Updated database with attached FPGAs: {json.dumps(host_machine_fpgas_db, indent=2)}")
+        
+        # Step 4: scp back while holding the lock
+        subprocess.run([
+            "ssh", host_machine_ip,
+            f"flock {lock_path} bash -c 'cat > {db_path}'"
+        ], input=open(local_tmp_path, "rb").read(), check=True)
+
+        # write to local firesim_vm_status.json
+        firesim_vm_status_path = pjoin(
+            os.path.dirname(os.path.abspath(__file__)), "..", "firesim_vm_status.json"
+        )
+        with open(firesim_vm_status_path, "r+") as f:
+            firesim_vm_status = json.load(f)
+            firesim_vm_status = {
+                "vm_setup": True,
+                "vm_ip": ip_addr,
+            }
+            f.seek(0)
+            json.dump(firesim_vm_status, f, indent=2)
+            f.truncate()
+    
+        # install cmake, gcc, etc
+        logging.getLogger("paramiko").setLevel(logging.DEBUG)
+        rootLogger.info(f"{self.vm_username}@{ip_addr}")
+        env.host_string = f"{self.vm_username}@{ip_addr}" # this changes the host_string for subsequent run() calls until program exits
+        env.key_filename = pjoin(
+            os.path.dirname(os.path.abspath(__file__)), "..", "vm-cloud-init-configs", "firesim_vm_ed25519"
+        )  # this is the key we use to ssh into the VM, set in vm-cloud-init-configs/user-data
+
+        # ssh into the vm is key based, set in vm-cloud-init-configs/user-data
+        rootLogger.info("Installing build-essential...")
+        run("""sudo apt-get install -y build-essential""", shell=True)
+
+        # install scripts to /usr/local/bin
+        rootLogger.info("Installing scripts to /usr/local/bin...")
+        run("""git clone https://github.com/firesim/firesim ~/firesim""", shell=True)
+        run("""cd firesim && sudo cp deploy/sudo-scripts/* /usr/local/bin""", shell=True)
+        run("""cd firesim && sudo cp platforms/xilinx_alveo_u250/scripts/* /usr/local/bin""", shell=True) # TODO: this is hardcoded to the u250 platform, need to make it more generic
+        run("""rm -rf ~/firesim""", shell=True) # remove the repo after copying the scripts
+
+        # install xdma & xcsec drivers
+        rootLogger.info("Installing xdma & xcsec drivers...")
+
+        run("""git clone https://github.com/Xilinx/dma_ip_drivers ~/dma_ip_drivers""", shell=True)
+
+        run("""cd ~/dma_ip_drivers/XDMA/linux-kernel/xdma && sudo make install""", shell=True)
+
+        run("""git clone https://github.com/paulmnt/dma_ip_drivers ~/dma_ip_drivers_xvsec""", shell=True) 
+
+        run("""cd ~/dma_ip_drivers_xvsec/XVSEC/linux-kernel && sudo make clean all && sudo make install""", shell=True)
```

Comment:
Woah, let's not execute Fabric this way, not only is it a bit confusing I suspect overriding `env` would apply globally. Instead do https://github.com/firesim/firesim/blob/e8c384a26737d667c909c0289e44564a18c54854/deploy/firesim#L310 like I mentioned before.